bin/pasta_maintainers_stats.py

```diff
@@ -0,0 +1,303 @@
+                        reverse=True)
+    results = []
+    for filename in filenames:
+        subsystems = all_maintainers.get_subsystems_by_file(filename)
+        subsystems -= {'THE REST'}
+        # To get a sense of how large the subsystem is and order by it, look at
+        # how many different patterns it includes:
+        # all_maintainers.subsystems[subsystem].files
+        subsystems = sorted(subsystems, 
+         key=lambda subsystem: len(all_maintainers.subsystems[subsystem].files))
+        results.append([filename, subsystems])
+    return results
+
+
+def maintainers_stats(config, sub, argv):
+    parser = ArgumentParser(description='Display file sizes grouped by '
+                                            'maintainers or subsystems')
+    parser.add_argument('show_maintainers', nargs=1, type=bool)
+    parser.add_argument('--entries', action='store_true',
+                            help='groups by subsystems')
+    parser.add_argument('--files', action='store_true',
+                            help='Displays files with corresponding entries')
+    parser.add_argument('--smallstat', action='store_true', help='Simple view')
+    parser.add_argument('--bytes', action='store_true', help='Show byte counts')
+    parser.add_argument('--outfile', type=str, help='Output to a csv file')
+    parser.add_argument('--filter', type=str, help='Filter by file list: '
+                        'enter the file name for the file containing the list '
+                        'of files to filter by.')
+    parser.add_argument('--revision', type=str, help='Specify a commit hash or '
+                        'a version name for a Linux repo')
+
+    args = parser.parse_args()
+    repo = config.repo
+
+    kernel_revision = 'HEAD'
+    if args.filter:
+        filter_by_files = file_to_string(args.filter).splitlines()
+        # The first line of the file must contain a valid kernel version,
+        # tag or commit hash
+        kernel_revision = filter_by_files.pop(0)
+
+    # arguments may override the kernel revision, if it is not already set
+    if args.revision:
+        kernel_revision = args.revision
+    log.info('Working on kernel revision %s' % kernel_revision)
+
+    all_maintainers = load_maintainer(repo, kernel_revision)
+    tree = get_tree(repo.repo, kernel_revision)
+    all_filenames = walk_commit_tree(tree)
+
+    global _tmp_tree
+    _tmp_tree = tree
+    processes = int(cpu_count())
+    p = Pool(processes=processes, maxtasksperchild=1)
+    result = p.map(get_file_size, all_filenames)
+    p.close()
+    p.join()
+    _tmp_tree = None
+
+    file_sizes = dict(result)
+
+    if args.entries:
+        title = ['Entries','LoC']
+        if args.filter:
+            results, irrelevant_dirs, relevant_dirs = get_counts_by_entry(
+                all_maintainers, file_sizes, filter_by_files)
+        else:
+            results, _, _= get_counts_by_entry(all_maintainers, file_sizes)
+    elif args.files:
```

Comment:
`--entries` and `--files` are mutual exclusive, right?

So if I call `maintainers-stats --entries --files`, then --files will by ignored.

In this case, it's probably better to make it clear by introducing a mode. E.g.
```
maintainers_stats --mode entries
maintainers_stats --mode files
```

We even have three modes, right? entries, files, and maintainers, if i see that correctly. We should make that clear in the commandline arguments.

And it's probably better to rename entries to subsystems.

Reply:
Yes, we should introduce some option like that, I would call it --group-by or --aggregate-on ... entries (or section, not subsystem)/files/maintainers instead of --mode.

Reply:
Yep, sounds good. And use, e.g., sections (formerly knows as entries) as default value.